### PR TITLE
removing setup remote docker from config.yml

### PR DESCRIPTION
**Pull request by hod-alpert**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Removing setup remote docker from all jobs but unittests since they are flaky and not needed.

* Added the `Not running this job` step because each job must have a least one step that runs in it in any case.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -316,15 +316,14 @@ references:
                 git fetch $USER $BRANCH
                 git checkout $USER/$BRANCH $PACK
                 exit 0
-    
+
 jobs:
   Setup Environment:
     <<: *container_config
     resource_class: medium+
```
Comment (glicht) [inline .circleci/config.yml:323]: Shouldn't this be `medium`?

File: .circleci/config.yml [hunk 2]
```diff
@@ -738,7 +787,20 @@ jobs:
...
     <<: *environment
     steps:
-      - setup_remote_docker
+      - when:
+          condition:
+            or:
+              - << pipeline.parameters.instance_tests >>
+              - << pipeline.parameters.contrib_branch >>
+              - << pipeline.parameters.demisto_sdk_nightly >>
+              - << pipeline.parameters.nightly >>
```
Comment (glicht) [inline .circleci/config.yml:796]: I think 6.0 should run in nightly. Not sure about non_ami_run. @yaakovi fyi

Comment (hod-alpert) [reply .circleci/config.yml:796]: My bad, you're right

Comment (hod-alpert) [reply .circleci/config.yml:796]: non_ami_run runs in 5_0 currently, fixing it too

**Comment by hod-alpert** [comment]
Non ami build example: https://app.circleci.com/pipelines/github/demisto/content/19589/workflows/5a02d056-9778-487c-aa7a-e8b4a6ba1899

**Pull request merged**